ethereum/executionlayer/src/test/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionLayerBlockProductionManagerImplTest.java

```diff
@@ -163,10 +169,13 @@ public void preDeneb_builderOnline() throws Exception {
     final SafeFuture<HeaderWithFallbackData> headerWithFallbackDataFuture =
         executionPayloadResult.getHeaderWithFallbackDataFuture().orElseThrow();
     assertThat(headerWithFallbackDataFuture.get()).isEqualTo(expectedResult);
+    final SafeFuture<UInt256> executionPayloadValueFuture =
+        executionPayloadResult.getExecutionPayloadValueFuture().orElseThrow();
+    assertThat(executionPayloadValueFuture.get()).isEqualTo(executionPayloadValue);
 
     // we expect both builder and local engine have been called
     verifyBuilderCalled(slot, executionPayloadContext);
-    verifyEngineCalled(executionPayloadContext, slot);
+    verifyEngineCalled(executionPayloadContext, slot, 2);
```

Comment:
it shouldn't be called for 2 times here and in the other cases, or may be I got it wrong. Let's clarify with @StefanBratanov

Reply:
Hmm yeah it should only be once I think. What is the reason of this change @mehdi-aouadi

Reply:
Done, added a callback when calling `builderGetHeader` in order to get `executionPayloadValue` at the same time and avoid calling the EL a second time

Reply:
Nice